Sources/DiscordKitBot/Client.swift

```diff
@@ -51,39 +81,109 @@ public final class Client {
     ///
-    /// > Warning: Ensure this function is called _before_ any calls to the API are made,
-    /// > and _after_ all event sinks have been registered. API calls made before this call
-    /// > will fail, and no events will be received while the gateway is disconnected.
+    /// > Important: This method will block the main thread for the rest of the execution time.
+    /// > This means that any code after this call in the main thread will never run, and should be considered unreachable.
+    /// > If you want to perform operations immediately after your bot is logged in, add a listener to ``ready``.
+    /// 
+    /// > Warning: Calling this method while a bot is already logged in will disconnect that bot and 
+    /// > replace it with the new one. You cannot have 2 bots logged in at the same time.
+    /// 
+    /// ## See Also
+    /// - ``login(filePath:)``
+    /// - ``login()``
     public func login(token: String) {
+        if Client.current != nil {
+            Client.current?.disconnect()
+        }
         rest.setToken(token: token)
         gateway = .init(token: token)
         evtHandlerID = gateway?.onEvent.addHandler { [weak self] data in
             self?.handleEvent(data)
         }
+        Client.current = self
+
+        // Handle exit with SIGINT
+        let signalCallback: sig_t = { signal in
+            print("Gracefully stopping...")
+            Client.current?.disconnect()
+            sleep(1) // give other threads a tiny amount of time to finish up
+            exit(signal)
+        }
+        signal(SIGINT, signalCallback)
+
+        if blockOnLogin {
+            // Block thread to prevent exit
+            RunLoop.main.run()
+        }
+
     }
+
+    /// Login to the Discord API with a token stored in a file
+    ///
+    /// This method attempts to retrieve the token from the file provided,
+    /// and calls ``login(token:)`` if it was found. Any API calls made before this method is ran will fail.
+    ///
+    /// > Important: This method will block the main thread for the rest of the execution time.
+    /// > This means that any code after this call in the main thread will never run, and should be considered unreachable.
+    /// > If you want to perform operations immediately after your bot is logged in, add a listener to ``ready``.
+    /// 
+    /// > Warning: Calling this method while a bot is already logged in will disconnect that bot and 
+    /// > replace it with the new one. You cannot have 2 bots logged in at the same time.
+    ///
+    /// - Parameter filePath: A path to the file that contains your Discord bot's token
+    ///
+    /// - Throws: `AuthError.emptyToken` if the file is empty.
+    ///
+    /// ## See Also
+    /// - ``login()``
+    /// - ``login(token:)``
+    public func login(filePath: String) throws {
+        let token = try String(contentsOfFile: filePath).trimmingCharacters(in: .whitespacesAndNewlines)
+        if token.isEmpty {
```

Comment:
Using a `guard` here might be more "Swifty".

```suggestion
        guard !token.isEmpty else {
```